astacus/coordinator/api.py

```diff
@@ -32,25 +36,40 @@ class OpName(str, Enum):
     cleanup = "cleanup"
 
 
-@router.get("/{op_name}/{op_id}")
-def op_status(*, op_name: OpName, op_id: int, c: Coordinator = Depends()):
-    op, op_info = c.get_op_and_op_info(op_id=op_id, op_name=op_name)
-    result = {"state": op_info.op_status}
-    if isinstance(op, (BackupOp, RestoreOp)):
-        result["progress"] = op.progress
-    return result
-
-
 class LockStartResult(Op.StartResult):
     unlock_url: str
 
 
 @router.get("/")
-def root():
+async def root():
     # Root is no-op, just useful for testing that Astacus is actually running
     return {}
 
 
+@router.post("/config/reload")
+async def config_reload(*, request: Request, c: Coordinator = Depends()):
+    """Reload astacus configuration"""
+    # First reload the config of the entire app (including coordinator and node)
+    config_path = os.environ.get("ASTACUS_CONFIG")
+    assert config_path is not None
+    config.set_global_config_from_path(request.app, config_path)
+    # Then clear the backup list cache
+    c.state.cached_list_response = None
+    if c.state.cached_list_running:
```

Comment:
This seems unnecessary. Couldn't the list backups endpoint just remember the config hash it started with, and check that at the end?

Reply:
It could and that would be more elegant, I'll change it, I don't like that bool either.

Reply:
In the end, I made the list backup endpoint remember the coordinator config instead of the hash of the source file.
Not a huge observable difference (mainly, the cache is busted less often, it's not busted if the config is equivalent).

The hash is useful in the "global" part and in the externally visible API because the external user deals with the config file.
But inside the list endpoint, we only care about the actual config and using the full hash was creating an annoying dependency forcing the app fixture for coordinator tests to add a fake hash.